src/admin.rs

```diff
@@ -311,7 +356,17 @@ where
     fn set_config(&mut self, input: &[u8]) -> Result<(), ConfigError> {
         let request: SetConfigRequest<'_> =
             cbor_deserialize(input).map_err(|_| ConfigError::DeserializationFailed)?;
-        config::set(&mut self.config, request.key, request.value)?;
+        let reset = config::set(&mut self.config, request.key, request.value)?;
+        if let Some(flag) = reset.signal {
+            flag.set_config_changed();
+        }
+        if let Some(client) = reset.client_id {
+            if let Err(_err) = syscall!(self.trussed.confirm_user_present(15 * 1000)).result {
+                debug_now!("Failed to verify user presence: {_err:?}");
+                return Err(ConfigError::NotConfirmed);
+            }
+            syscall!(self.trussed.factory_reset_client(client));
+        }
```

Comment:
Maybe we can use the following API:

```rust
pub trait Config: Default + PartialEq + DeserializeOwned + Serialize {
    fn field(&mut self, key: &str) -> Option<ConfigValueMut<'_>>;

    /// Client ID to factory-reset if the associated configuration option is changed
    ///
    /// # If the Request is for a `client_id`:
    ///
    /// - MUST return `Some` to indicate that the client can be factory reset by the admin app,
    /// - MUST return None otherwise.
    fn reset_client_id(
        &self,
        _key: &str,
    ) -> Option<(&'static Path, &'static ResetSignalAllocation)> {
        None
    }
}
```

Reply:
Sounds reasonable.

Reply:
Theoretically, there could be use cases where we want to detect config changes without performing a factory reset, but maybe we should keep things simple for the moment.

Reply:
> Theoretically, there could be use cases where we want to detect config changes without performing a factory reset, but maybe we should keep things simple for the moment.

For that we probably would also want the information of which config changed to what value.